web/src/app/rotations/components/RotationSetFavoriteButton.js

```diff
@@ -0,0 +1,77 @@
+import React from 'react'
+import p from 'prop-types'
+import gql from 'graphql-tag'
+import { graphql2Client } from '../../apollo'
+import Query from '../../util/Query'
+import { Mutation } from 'react-apollo'
+import SetFavoriteButton from '../../util/SetFavoriteButton'
+
+const query = gql`
+  query favQuery($id: ID!) {
+    rotation(id: $id) {
+      id
+      isFavorite
+    }
+  }
+`
+
+const mutation = gql`
+  mutation setFav($input: SetFavoriteInput!) {
+    setFavorite(input: $input)
+  }
+`
+
+export default class RotationSetFavoriteButton extends React.PureComponent {
```

Comment:
@m17ch @arurao @mastercactapus Should a generic `SetFavoriteButton` include both the query and the mutations, or just include the icon/relevant styling?